src/operand/reference.ts

```diff
@@ -100,6 +101,18 @@ export class Reference extends Operand {
     return contextValueLookup(ctx, this.key)
   }
 
+  simplify (ctx: Context): Result | Evaluable {
+    const key = this.key.split(/\./)[0]
+    if (ctx[key] !== undefined) { // TODO excluded ids
+      return this.evaluate(ctx)
+    }
+    return this
+  }
+
+  serialize ({ referenceSerialization }: Options): string {
+    return referenceSerialization(this.key)
```

Comment:
OMG this is much much much more complex than just this.

Reply:
if needed some insight, @lilyvanoekel would be the best resources, as she's implemented the nested references, composite references.